src/components/ImageSettingsModal.jsx

```diff
@@ -103,8 +104,8 @@ export default class ImageSettingsModal extends Component {
               />
             </div>
             <div className={elementStyles.modalButtons}>
-              <button type="button" className={errorMessage ? elementStyles.disabled : elementStyles.blue} disabled={!!errorMessage} onClick={this.renameImage}>Save</button>
-              <button type="button" className={sha ? elementStyles.warning : elementStyles.disabled} onClick={this.deleteImage} disabled={!sha}>Delete</button>
+              <button type="button" className={errorMessage ? elementStyles.disabled : elementStyles.blue} disabled={!!errorMessage} onClick={_.throttle(this.renameImage, 2000)}>Save</button>
```

Comment:
Also I know this won't be relevant if we decide against using throttle, but a note on style: we should define magic numbers like `2000` as a constant at the top of the file and re-use it across the file.

Reply:
yea definitely